manager/orchestrator/replicatedjob/orchestrator.go

```diff
@@ -45,6 +52,166 @@ func (o *Orchestrator) Run(ctx context.Context) {
 	<-o.stopChan
 }
 
+// reconcileService reconciles the replicated job service with the given ID by
+// checking to see if new replicas should be created. reconcileService returns
+// an error if there is some case prevent it from correctly reconciling the
+// service.
+func (o *Orchestrator) reconcileService(id string) error {
+	var (
+		service *api.Service
+		tasks   []*api.Task
+	)
+	// first, get the service and all of its tasks
+	o.store.View(func(tx store.ReadTx) {
+		service = store.GetService(tx, id)
+
+		// getting tasks with FindTasks should only return an error if we've
+		// made a mistake coding; there's no user-input or even reasonable
+		// system state that can cause it. If it returns an error, we'll just
+		// panic and crash.
```

Comment:
Not 100% sure I agree with this: it might very well be the case as of right now that `FindTasks` can never return an error, but what if some change makes it reasonable for it to start returning errors? Then if we have panics all over our code base for that, that's going to be a major pain to change.

Besides, it doesn't seem ideal to rely on your deep system knowledge of what functions are reasonably going to error out or not: I think that it either should be made obvious by the API (by panicking in `FindTasks` itself if there is an error, but never returning one), or else errors out of `FindTasks` should just be treated normally.

Reply:
I'd argue that we should document and lock down `FindTasks` as returning only errors in the certain rather-catastrophic cases that it currently does, but still defer the actual panicking (if desired) to the API consumer.

That said, here, we can just return an error. I had some code, now removed because it was out of scope for this particular commit, that panicked during orchestrator initialization, which would have been more appropriate, as Run can return no errors. Basically, during initialization, if we cannot retrieve the objects needed to initialize the state of the Orchestrator, there isn't a reasonable action we can take to continue, and we need to panic, crash the daemon, and consequently force a leadership change to try again.

However, we don't need to argue that here yet. I'll just change this to return an error in the case that this find fails.